packages/apollo-link-state/src/index.ts

```diff
@@ -83,6 +87,8 @@ export const withClientState = (
       };
 
       return new Observable(observer => {
+        const clientQuery = getClientSetsFromDocument(operation.query);
+        const clientData = cache && cache.readQuery({ query: clientQuery });
```

Comment:
@evans @peggyrayzis readQuery in line 91 is causing the tests to fails.

Reply:
This client query should be done in the `graphql` call, try changing the query argument [here](https://github.com/apollographql/apollo-link-state/pull/193/files/d48b1c9738c6abfcc5a1add6065b51166ea5025d#diff-cda1c9af8b0b40f97820e2a76de23040R106) to `clientQuery`

Reply:
The reason for the failure is due to the cache being empty on the first read. The client-side data is populated by the `graphql` call. If you could add a comment above that call saying that this is where the client data is fetched, that would be incredible!